client/state/selectors/get-checkout-incompatible-products.ts

```diff
@@ -0,0 +1,82 @@
+/**
+ * External dependencies
+ */
+import { translate, TranslateResult } from 'i18n-calypso';
+
+/**
+ * Internal dependencies
+ */
+import { DefaultRootState } from 'react-redux';
```

Comment:
Perhaps use this instead? Regardless, this should be a `type` import.
```suggestion
import type { AppState } from 'types';
```

Reply:
Done!